Wangscape/OptionsManager.cpp

```diff
@@ -0,0 +1,54 @@
+#include "OptionsManager.h"
+
+#include <fstream>
+
+#include <boost/filesystem.hpp>
+
+#include <spotify/json.hpp>
+
+#include "codecs/OptionsCodec.h"
+
+OptionsManager::OptionsManager(std::string filename)
+{
+    std::ifstream ifs(filename);
+    if (!ifs.good())
+    {
+        throw std::runtime_error("Could not open options file");
+    }
+
+    std::string str{std::istreambuf_iterator<char>(ifs),
+                    std::istreambuf_iterator<char>()};
+
+    mOptions = spotify::json::decode<Options>(str.c_str());
+    mOptions.filename = filename;
+
+    // TODO(hryniuk): move it elsewhere
+    auto outputDirectory = mOptions.outputDirectory;
```

Comment:
Hmm, yes, you're right. Creating the output directory shouldnt be part of `loadOptions`, or `OptionsManager`. But if creating that directory fails, the error should be detected before any images are loaded or generated. The exception would be in eg. test cases where options must be loaded but output will not be written.

In any case, the `loadOptions` code shouldn't be duplicated anywhere.